controller/proxy-injector/server.go

```diff
@@ -76,24 +77,17 @@ func (w *WebhookServer) Shutdown() error {
 	return w.Server.Shutdown(context.Background())
 }
 
-func tlsConfig(certFile, keyFile string) (*tls.Config, error) {
-	certBytes, err := ioutil.ReadFile(certFile)
-	if err != nil {
-		return nil, err
-	}
-
-	keyBytes, err := ioutil.ReadFile(keyFile)
-	if err != nil {
-		return nil, err
-	}
+func tlsConfig(certFile, keyFile string, rootCA *ca.CA) (*tls.Config, error) {
+	dnsName := "linkerd-proxy-injector.linkerd.svc"
+	certAndPrivateKey, err := rootCA.IssueEndEntityCertificate(dnsName)
 
-	certPEM, err := pem.PEMEncodeCert(certBytes)
+	certPEM, err := pem.PEMEncodeCert(certAndPrivateKey.Certificate)
```

Comment:
Right, per my comment above, maybe we can provide the functionality of `PEMEncodeCert` and `PEMEncodeKey` funcs directly from the CA itself.